ui/app/src/utils/content.ts

```diff
@@ -950,3 +954,56 @@ export function getComputedPublishingTarget(item: DetailedItem): PublishingTarge
       ? 'staging'
       : null;
 }
+
+export function cleanupFolderName(name: string, allowBraces: boolean = false): string {
+  let cleanedUpName = slugify(name, {
+    // Setting `strict: true` would disallow `_`, which we don't want.
+    strict: false,
+    // Because of the moment where the library trims, `trim: true` caused undesired replacement of `-`
+    // at the beginning or end of the slug.
+    trim: false
+  });
+  return cleanedUpName.replace(allowBraces ? /[^a-zA-Z0-9-_{}]/g : /[^a-zA-Z0-9-_]/g, '');
+}
+
+export function cleanupAssetName(name: string, allowBraces: boolean = false): string {
+  return name.replace(allowBraces ? /[^a-zA-Z0-9-_{}.]/g : /[^a-zA-Z0-9-_.]/g, '').replace(/\.{1,}/g, '.');
+}
+
+export const getExtension = (type: string, name?: string) => {
+  if (type === 'asset') {
+    const regexFileExtension = /[^\\]*\.(\w+)$/;
+    return name?.match(regexFileExtension)[1];
+  } else {
+    return type === 'controller' ? `groovy` : `ftl`;
+  }
+};
+
+export const getName = (type: string, name: string) =>
+  `${name}.${getExtension(type)}`.replace(/(\.groovy)(\.groovy)|(\.ftl)(\.ftl)/g, '$1$3').replace(/\.{2,}/g, '.');
```

Comment:
Same as above: used in multiple places; leave in its original location if not? Needs better name.

Reply:
renamed to getFileNameWithExtensionForItemType